### ISSUE TITLE
overloads without implementations (in stub files) are not detected

### ISSUE THREAD

**Issue by tlambert03** [mkdocstrings/griffe#116]
**Describe the bug**
Methods and functions that are decorated with `@overload` but do not have an implementation (as is standard in a stub file) are not detected:

**To Reproduce**
```python
# mystub.pyi
from typing import overload

class M:
    @overload
    def f(self, x: int) -> int: ...
    @overload
    def f(self, x: str) -> str: ...

class M2:
    @overload
    def f(self, x: int) -> int: ...
    @overload
    def f(self, x: str) -> str: ...
    def f(self, x: str | int) -> str | int: ...  # one wouldn't normally do this in a stub
```

```python
In [3]: griffe.load('mystub._test.M').members
Out[3]: {}

In [4]: griffe.load('mystub._test.M2').members
Out[4]: {'f': <Function('f', 14, 14)>}
```

**Expected behavior**
Expected `f` method to be detected.

**System (please complete the following information):**
- `griffe` version: 0.24.1
- Python version: 3.10
- OS: mac

Will probably be relevant for https://github.com/mkdocstrings/mkdocstrings/issues/308 as well

**Comment by pawamoy** [comment]
Thanks for the report!
It's maybe a bug in the code that merges stubs data into actual module data, but I can't see it :thinking: 
Do you have a public repo I could use to debug?

**Comment by tlambert03** [comment]
the repo i found it on was that same (compiled) [`pymmcore`](https://github.com/micro-manager/pymmcore) repo.  so

```
pip install pymmcore
```

```python
def test_overloads():
    import griffe
    core = griffe.load('pymmcore.CMMCore')
    assert 'defineConfig' in core.members

    # this is a non-overloaded function that works already
    assert 'addSearchPath' in core.members
```

and the corresponding stub is at https://github.com/micro-manager/pymmcore/blob/main/pymmcore/__init__.pyi

**Comment by pawamoy** [comment]
Ah, found the issue. It's because we merge stubs before wildcard imports were expanded.

**Comment by pawamoy** [comment]
Another issue is that `./pymmcore` is found first, rather than the installed version which is the only one containing the compiled module `_pymmcore_swig.cpython-311-x86_64-linux-gnu`. So the compiled module is not found, not inspected, and we can't expand wildcards, and merge stubs. An argument in favor of the `src`-layout I guess. I'll try with it, to see if it's better.

**Comment by pawamoy** [comment]
OK, we need https://github.com/mkdocstrings/griffe/pull/115 first.

**Comment by pawamoy** [comment]
Fixed in 0ed9c36
